### PR TITLE
Refactor:  separate UI from game logic

### PR DESCRIPTION
**Pull request by FlashGooden**
## Description
We were tasked with separating the game Logic from UI logic. I am submitting a draft to help us get on task.

## Spec

See Issue: [ISSUE_NUMBER](https://sparkbox.atlassian.net/browse/FSA2021-68)

## Validation

- [ ] This PR has code changes, and our linters still pass.
- [ ] This PR has new code, so new tests were added or updated, and they pass.

### To Validate

1. Make sure all PR Checks have passed.
2. Pull down all related branches.
3. Navigate to `refactor--seperate-UI-from-game-logic`
3. run `npm start`
4. go to `localhost:3000`
5. interact with game UI the same way

### CODE REVIEW

**Review by rise-erpelding — COMMENTED** [review]
🤯 Is this the thing that was keeping us from being able to separate all of the js code into multiple files?

File: .eslintrc.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ module.exports = {
   root: true,
   env: {
     node: true,
+    browser: true,
```
Comment (rise-erpelding) [inline .eslintrc.js:5]: 🤯 Is this the thing that was keeping us from being able to separate all of the js code into multiple files?

File: src/js/index.spec.js [hunk 2]
```diff
@@ -1,13 +1,9 @@
 import jsdom from 'jsdom';
-import handleClickCell from './Game/game';
+import { handleClickCell, game } from './Game/game';
```
Comment (rise-erpelding) [inline src/js/index.spec.js:2]: > As far as I can tell, it only works if you import the actual variable, as the handleClickCell function uses the game variable directly from the game.js file.

Makes sense to me why you would do it this way.

File: .eslintrc.js (hunk 1 above)
Comment (FlashGooden) [reply .eslintrc.js:5]: I think this just lets when we call 'document' in our index.js not have a linting issue. There was a separate linting rule which was messing up our imports. We need for right now to explicitly run the import with '.js' file at the end until we figure out how to mayyyyybe change the babel configuration?

File: src/js/index.spec.js (hunk 2 above)
Comment (FlashGooden) [reply src/js/index.spec.js:2]: Yea.... I think eventually we can import state. I'm just going with what works for right now.

File: .eslintrc.js (hunk 1 above)
Comment (rise-erpelding) [reply .eslintrc.js:5]: I didn't even catch the .js on the end, I'm glad you mentioned that! Ok, so maybe _that_ was the thing.

**Review by rise-erpelding — APPROVED** [review]
Looks great! Tests pass, everything works, and we've incorporated the changes from main. Green light!

**Pull request merged**